### PR TITLE
 Allow to reorganize field order in fromatter dialog

### LINKED ISSUE #4379: `xml-editor`: Add support for rearranging items in a table format
**Describe the bug**
You can add items to a table format but you cannot remove them. To change the order, you are required to remove everything and add them in a different order.

**To Reproduce**
1. Create or edit a table format
2. Add fields
3. See issue 

https://kufishdemo-xml-editor.test.specifysystems.org/specify/resources/app-resource/9/formatter/Loan/39

https://github.com/specify/specify7/assets/37256050/0f5788ed-e52f-466a-870a-5aecda7827f8

macOS, Arc

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4379 

### Checklist

- [x] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [x] Add relevant issue to release milestone

### Testing instructions

- open app resources 
- open a formatter 
- add several fields 
- reorganize the fields order by using the up and own arrows

### CODE REVIEW

**Comment by CarolineDenis** [comment]
https://github.com/specify/specify7/assets/108160931/9c50d6d3-2847-4c71-8904-3af9240c5146

File: specifyweb/frontend/js_src/lib/components/Formatters/Fields.tsx [hunk 1]
```diff
@@ -64,6 +78,20 @@ export function Fields({
                   (field): void => setFields(replaceItem(fields, index, field)),
                 ]}
                 key={index}
+                moveFieldDown={
+                  index === fields.length - 1
+                    ? undefined
+                    : (): void => {
+                        moveField(index, 'down');
+                      }
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/Formatters/Fields.tsx:86]:
```suggestion
                    : (): void => moveField(index, 'down')
```

File: specifyweb/frontend/js_src/lib/components/Formatters/Fields.tsx [hunk 2]
```diff
@@ -29,6 +30,19 @@ export function Fields({
   readonly fields: GetSet<Formatter['definition']['fields'][number]['fields']>;
 }): JSX.Element {
   const [displayFormatter, setDisplayFormatter] = React.useState(false);
+
+  const moveField = (index: number, direction: 'down' | 'up'): void => {
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/Formatters/Fields.tsx:34]:
```suggestion
function moveField (index: number, direction: 'down' | 'up'): void {
```

File: specifyweb/frontend/js_src/lib/components/Formatters/Fields.tsx [hunk 3]
```diff
@@ -29,6 +30,19 @@ export function Fields({
...
+    const updatedFields = Array.from(fields);
+    const targetIndex = direction === 'up' ? index - 1 : index + 1;
+
+    [updatedFields[index], updatedFields[targetIndex]] = [
+      updatedFields[targetIndex],
+      updatedFields[index],
+    ];
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/Formatters/Fields.tsx:41]: oh, I forgot this was possible - cool!

**Review by maxpatiiuk — APPROVED** [review]
this looks good, but I though you were going to use the draggable library that you have in the query builder

**Review by grantfitzsimmons — APPROVED** [review]

**Review by pashiav — APPROVED** [review]
Looks good!

https://github.com/specify/specify7/assets/118556884/2b826748-0b78-419c-84ed-042383c74d35

**Pull request merged**